expr/func_aliasbymetric.go

```diff
@@ -0,0 +1,36 @@
+package expr
+
+import (
+	"github.com/grafana/metrictank/api/models"
+)
+
+type FuncAliasByMetric struct {
+	in GraphiteFunc
+}
+
+func NewAliasByMetric() GraphiteFunc {
+	return &FuncAliasByMetric{}
+}
+
+func (s *FuncAliasByMetric) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+	}, []Arg{ArgSeries{}}
+}
+
+func (s *FuncAliasByMetric) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncAliasByMetric) Exec(dataMap DataMap) ([]models.Series, error) {
+	series, err := s.in.Exec(dataMap)
+	if err != nil {
+		return nil, err
+	}
+	for i, serie := range series {
+		n := extractMetric(serie.Target)
```

Comment:
This is not what aliasByMetric does.
aliasByMetric takes the "base" of the metric as the alias (the last word)

e.g. `foo.bar.baz` -> `baz`
`perSecond(foo.bar.baz)` -> `baz`

extractMetric is used to extract full metric names.
so I think you should just keep `n := extractMetric(serie.Target)`, then find the last occurence of `.` within `n` and keep the string after that last `.`

Reply:
Thanks for the clarification. With the graphite implementation, the pattern we were seeing is if the metric contains tags with a period in them like `foo.bar.baz;tag1=one.blah;tag2=twoblah` what would be returned is `blah;tag2=twoblah` so everything after the last period is encountered. That's wrong and needs to be fixed in this implementation, correct?

Reply:
yes, the behavior which you describe definitely sounds like a bug and isn't intended

Reply:
Roger that. I've made the required changes of extracting only the base name.
Now, the expected behavior is:
- `perSecond(foo.bar.baz)` -> `baz`
- `perSecond(foo.bar.baz;tag1=one.blah;tag2=twoblah)` -> `baz`
Please let me know if this is correct.

I'll re-request a review after testing it throughly on my end.

Reply:
@vaguiar I recommend you use a graphite server to test against and see what the behavior is. (notwithstanding the bug we highlighted). When I test something like `aliasByMetric(perSecond(foo.bar.baz;tag1=one.blah;tag2=twoblah))` or just `aliasByMetric(foo.bar.baz;tag1=one.blah;tag2=twoblah)`, the output is `baz;tag1=one.blah;tag2=twoblah`

Reply:
Noted. I ran tests to compare the graphite output with that of metrictank. Thanks for the patient clarifications and apologies for the back and forth.